src/main/java/com/google/cloud/resourcemanager/LienInfo.java

```diff
@@ -0,0 +1,210 @@
+ */
+public class LienInfo {
+
+  static final Function<Lien, LienInfo> FROM_PROTOBUF_FUNCTION =
+      new Function<Lien, LienInfo>() {
+        @Override
+        public LienInfo apply(Lien lienProtobuf) {
+          return LienInfo.fromProtobuf(lienProtobuf);
+        }
+      };
+  static final Function<LienInfo, Lien> TO_PROTOBUF_FUNCTION =
+      new Function<LienInfo, Lien>() {
+        @Override
+        public Lien apply(LienInfo lienInfo) {
+          return lienInfo.toProtobuf();
+        }
+      };
+
+  private String createTime;
+  private String name;
+  private String origin;
+  private String parent;
+  private String reason;
+  private List<String> restrictions;
+
+  /** Builder for {@code LienInfo}. */
+  static class Builder {
+
+    private String createTime;
+    private String name;
+    private String origin;
+    private String parent;
+    private String reason;
+    private List<String> restrictions;
+
+    Builder(String parent) {
+      this.parent = parent;
+    }
+
+    Builder(LienInfo info) {
+      this.createTime = info.createTime;
+      this.name = info.name;
+      this.origin = info.origin;
+      this.parent = info.parent;
+      this.reason = info.reason;
+      this.restrictions = info.restrictions;
+    }
+
+    public Builder setCreateTime(String createTime) {
+      this.createTime = createTime;
+      return this;
+    }
+
+    public Builder setName(String name) {
+      this.name = name;
+      return this;
+    }
+
+    public Builder setOrigin(String origin) {
+      this.origin = origin;
+      return this;
+    }
+
+    public Builder setReason(String reason) {
+      this.reason = reason;
+      return this;
+    }
+
+    public Builder setRestrictions(List<String> restrictions) {
+      this.restrictions = restrictions;
+      return this;
+    }
+
+    public Builder setParent(String parent) {
+      this.parent = checkNotNull(parent);
+      return this;
+    }
+
+    public LienInfo build() {
+      return new LienInfo(this);
+    }
+  }
+
+  LienInfo(Builder builder) {
+    this.createTime = builder.createTime;
+    this.name = builder.name;
+    this.origin = builder.origin;
+    this.parent = builder.parent;
+    this.reason = builder.reason;
+    this.restrictions = builder.restrictions;
+  }
+
+  /** Returns the creation time of the lien. */
+  public String getCreateTime() {
+    return createTime;
+  }
+  /** Returns the name of the lien. */
+  public String getName() {
+    return name;
+  }
+  /** Returns the origin of the lien */
+  public String getOrigin() {
+    return origin;
+  }
+  /** Returns the parent of the lien. */
+  public String getParent() {
+    return parent;
+  }
+  /** Returns the reason of the lien. */
+  public String getReason() {
+    return reason;
+  }
+  /** Returns the restrictions of the lien. */
+  public List<String> getRestrictions() {
+    return restrictions;
```

Comment:
This exposes potentially mutable internal state so making a copy is a good idea.

Reply:
This method shouldn't return an immutable list when it's declared to return a list.

Reply:
done